sqlglot/dialects/duckdb.py

```diff
@@ -739,6 +739,9 @@ class Generator(generator.Generator):
             exp.Lateral: explode_to_unnest_sql,
             exp.LogicalOr: rename_func("BOOL_OR"),
             exp.LogicalAnd: rename_func("BOOL_AND"),
+            exp.Lower: lambda self, e: f"CAST({self.func('LOWER', e.this)} AS BLOB)"
+            if isinstance(e.this, exp.ByteString)
+            else self.func("LOWER", e.this),
```

Comment:
We shouldn't just check for `ByteString`– this only handles cases like `LOWER(b'...')`. What happens though if the input is `LOWER(bytes_col)`?

There's a pattern that we use for type-sensitive transpilations where you do:

```python
if not expression.type:
    from sqlglot.optimizer.annotate_types import annotate_types

    expression = annotate_types(expression, dialect=self.dialect)

if expression.is_type(...):
    # Handle transpilation for a set of type(s) here
```

This handles both `ByteString` and other cases, so it's a more robust and general solution. We'll need to define `lower_sql` within `DuckDB.Generator` instead of having a `TRANSFORMS` entry.